includes/class-wcpdf-settings-upgrade.php

```diff
@@ -98,11 +98,90 @@ public function extension_overview( $tab, $section ) {
 					'extensions' => ['templates', 'bundle'],
 				),
 			);
+			
+			$transient_name = 'wpo_wcpdf_extension_license_infos';
+			//delete_transient( $transient_name ); for debug only
+			if ( false === ( $extension_license_infos = get_transient( $transient_name ) ) ) {
+				$extension_license_infos = [];
+				foreach ( [ 'pro', 'templates' ] as $extension ) {
+					$extension_license_infos[$extension] = $this->get_extension_license_info( $extension );
+				}
+				
+				set_transient( $transient_name, $extension_license_infos, DAY_IN_SECONDS );
+			}
 
 			include( WPO_WCPDF()->plugin_path() . '/includes/views/upgrade-table.php' );
 		}
 	}
 	
+	/**
+	 * Check if a PDF extension is enabled
+	 *
+	 * @param  string  $extension  can be 'pro' or 'templates'
+	 * @return boolean
+	 */
+	public function extension_is_enabled( $extension ) {
+		$is_enabled = false;
+		
+		if ( ! empty( $extension ) || ! in_array( $extension, [ 'pro', 'templates' ] ) ) {
+			$extension_main_function = "WPO_WCPDF_".ucfirst( $extension );
+			if ( function_exists( $extension_main_function ) ) {
+				$is_enabled = true;
+			}
+		}
+		
+		return $is_enabled;
+	}
+	
+	/**
+	 * Get PDF extension license info
+	 *
+	 * @param  string  $extension  can be 'pro' or 'templates'
+	 * @return array
+	 */
+	public function get_extension_license_info( $extension ) {
+		$license_info = [
+			'enabled' => false
+		];
+		
+		if ( $this->extension_is_enabled( $extension ) ) {
+			$license_info['enabled'] = true;
+			$extension_main_function = "WPO_WCPDF_".ucfirst( $extension );
+			
+			if ( $extension == 'templates' && version_compare( $extension_main_function()->version, '2.20.0', '<=' ) ) { // 'updater' property had 'private' visibility
+				return $license_info;
+			}
+			
+			if ( is_callable( [ $extension_main_function()->updater, 'remote_license_actions' ] ) ) {
+				if ( ! empty( $license_key = $extension_main_function()->updater->get_license_key() ) ) {
+					$args = array(
+						'edd_action'  => 'check_license',
+						'license_key' => trim( $license_key ),
+					);
+				} else {
+					$args = array();
+				}
+				
+				$request = $extension_main_function()->updater->remote_license_actions( $args );
+				if ( is_object( $request ) && isset( $request->license ) ) {
+					$license_info['status']         = $request->license;
+					$license_info['bundle_upgrade'] = ! empty( $request->lowest_bundle_upgrade ) && is_string( $request->lowest_bundle_upgrade ) ? $request->lowest_bundle_upgrade : ''; // https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/503#issuecomment-1523066696
```

Comment:
This might cause an issue with users that have a 3-site or 25-site license for one of the extensions. Because I think it will return a 1-site Bundle upgrade link.